### PR TITLE
Add support for Oracle database plugin connection options

### PR DESCRIPTION
**Pull request by dheeg**
### Description

In 2021 the Oracle database plugin for Vault introduced new configuration options as described here: https://github.com/hashicorp/vault-plugin-database-oracle/pull/62

This PR exposes those database connection options, which are especially important for Oracle AWS RDS installations, as those don't support the default plugin revocation statements see: https://github.com/hashicorp/vault-plugin-database-oracle/issues/45

I personally don't expect that this PR will receive a lot of upvotes, because it's a niche scenario => Oracle on AWS + Vault + Terraform to provision Vault. However it might be important for a few enterprises (including us).

### Checklist
- [x] Added [CHANGELOG](https://github.com/hashicorp/terraform-provider-vault/blob/master/CHANGELOG.md) entry (only for user-facing changes)
- [x] Acceptance tests where run against all supported Vault Versions

### Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

### CODE REVIEW

**Comment by dheeg** [comment]
@fairclothjm Sorry for pinging directly. Is there a chance to get a review for this PR?

**Review by fairclothjm — COMMENTED** [review]
@dheeg Hello, thanks for the contribution! Sorry for the delay. Could you rebase your changes on the main branch? Thanks!

**Comment by ram-parameswaran** [comment]
> @dheeg Hello, thanks for the contribution! Sorry for the delay. Could you rebase your changes on the main branch? Thanks!

Completed on behalf of user @dheeg

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ FEATURES:
 FEATURES:
 * Add support for `ext_key_usage_oids` in `vault_pki_secret_backend_role` ([#2108](https://github.com/hashicorp/terraform-provider-vault/pull/2108))
 * Adds support to `vault_gcp_auth_backend` for common backend tune parameters ([#1997](https://github.com/terraform-providers/terraform-provider-vault/pull/1997)).
+* Add support for Oracle database plugin configuration options `split_statements` and `disconnect_sessions`: ([#2085](https://github.com/hashicorp/terraform-provider-vault/pull/2085))
```
Comment (fairclothjm) [inline CHANGELOG.md:11]: This should go under the Unreleased section.

Comment (ram-parameswaran) [reply CHANGELOG.md]: @fairclothjm done, moved the changelog entry

**Review by fairclothjm — APPROVED** [review]
LGTM, Thanks!

**Pull request merged**